File: Daily Challanges/2026-07-July/2026-07-04.py
def kaprekar(number):
    count = 0

    while number != 6174:
        digits = f"{number:04d}"
        large = int("".join(sorted(digits, reverse=True)))
        small = int("".join(sorted(digits)))
        number = large - small
        count += 1

    return count


print(kaprekar(1234))
print(kaprekar(2025))
print(kaprekar(7173))
print(kaprekar(3164))
print(kaprekar(8082))

File: Daily Challanges/2026-07-July/2026-07-04.js
function kaprekar(number) {
  let count = 0;

  while (number !== 6174) {
    let digits = number.toString().padStart(4, "0").split("");
    let large = parseInt([...digits].sort((a, b) => b - a).join(""));
    let small = parseInt([...digits].sort((a, b) => a - b).join(""));
    number = large - small;
    count++;
  }

  return count;
}

console.log(kaprekar(1234));
console.log(kaprekar(2025));
console.log(kaprekar(7173));
console.log(kaprekar(3164));
console.log(kaprekar(8082));
